CMakeLists.txt

```diff
@@ -5,29 +5,41 @@ message(STATUS "Building using CMake version: ${CMAKE_VERSION}")
 
 project(S2Geography)
 
-set(CMAKE_CXX_STANDARD 11)
-set(CMAKE_CXX_STANDARD_REQUIRED True)
+set(CMAKE_CXX_STANDARD 17 CACHE STRING "The C++ standard to build with")
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
-set(ABSL_ENABLE_INSTALL ON)
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/third_party/cmake")
+
+option(S2GEOGRAPHY_FETCH_S2GEOMETRY "Download and build s2geometry from source" OFF)
```

Comment:
Having this option ON by default is fine by me, even tough I tend to see `find_package()` used by default in many C++ libraries.

I would keep this option, though. I think it's better to have a cmake error message when S2 is not found instead of implicitly downloading it (which might pick another version than the one expected).

Reply:
Ah yes I didn't how Arrow's CMake build scripts were organized but it looks nice indeed. Let's add a `s2_SOURCE` option with `AUTO` (default) / `BUNDLE` / `SYSTEM` / .. values then?

Reply:
Yes, certainly a bit overkill for S2Geography, but I like it, it should make everyone happy :-). I think I'm going to shamelessly copy some cmake lines from Arrow.